src/System.Private.Xml/src/System/Xml/Core/XmlReaderSettings.cs

```diff
@@ -125,7 +125,7 @@ internal XmlResolver GetXmlResolver()
         //notice we must keep GetXmlResolver() to avoid dead lock when init System.Config.ConfigurationManager
         internal XmlResolver GetXmlResolver_CheckConfig()
         {
-            if (!IsXmlResolverSet)
+            if (!LocalAppContextSwitches.DontProhibitDefaultResolver && !IsXmlResolverSet)
```

Comment:
We need to check for the value of this switch to give the option to use `<xsd:import schemaLocation"..." />` to import external xsd inside another one. This is not allowed by default.